### PR TITLE
Update localizations and schemes.

### PR DESCRIPTION
**Pull request by HechtiDerLachs**
Some updates and extensions regarding the localizations and schemes. 

This makes use of @thofma 's new code on polynomial maps.

### CODE REVIEW

File: experimental/Schemes/AffineSchemes.jl [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+export EmptyScheme
+
+export is_open_embedding, is_closed_embedding, canonically_isomorphic, hypersurface_complement, subscheme, name_of, set_name!
 export closure, product
 
-export SpecMor
-export pullback, domain, codomain, preimage, restrict, graph
+export SpecMor, morphism_type
+export pullback, domain, codomain, preimage, restrict, graph, identity_map, inclusion_map, is_isomorphism, is_inverse_of
+
```
Comment (thofma) [inline experimental/Schemes/AffineSchemes.jl:16]:
```suggestion
# TODO for Tommy: Find out why the following are necessary
```

**Review by thofma — COMMENTED** [review]
Minor comment on a name. Since we have `isisomorphic`, I would suggest to change `canonically_isomorphic` to `is[_]canonically_isomorphic`.

**Comment by HechtiDerLachs** [comment]
It seems that there is a conflict with the plane curves about the definition of `ProjectiveScheme`. These files are still in the `experimental` folder. 
- Who is using them? 
- Is anyone still developing this? Who can I talk to about making the two patches coherent? 
- If there's nobody available for the latter: How important is it that the projective spaces from the plane curves are being kept? 
CC: @wdecker

**Comment by HechtiDerLachs** [comment]
This PR is already quite big again, but it can be split in two main patches:

 * the update on existing code in Oscar up to [this commit](https://github.com/oscar-system/Oscar.jl/pull/1110/commits/cefad779891c6d630f812b9c8f5e5d9048e53348)
 * some more extensions on glueings and projective schemes.

If desired, we could therefore split it into two PRs. Let me know if this would be better.

**Comment by thofma** [comment]
I can't say much about the scheme stuff. From the map point of view of this looks good. Did you see my comment about `is[_]canonically_isomorphic`?

**Comment by HechtiDerLachs** [comment]
Ok, thanks for having a look at it! 

Yes, I saw your comment and I'm probably going to change it. We were not very happy with the name, anyway, so if there are any suggestions for something shorter, let me know.

BTW: I'm just running some performance tests and the `hom` in `MPolyRing.jl`, line 25, seems to be a bottleneck. Any ideas why?

**Comment by thofma** [comment]
How to reproduce it? I can check out this branch here, but what should I run?

**Comment by thofma** [comment]
Do you want to rename or should I just merge it as is?

**Comment by HechtiDerLachs** [comment]
Sorry, but I'm afraid, I should probably have another look at this PR. At some points (outside the `MPoly(Quo)LocalizedRingHom`s) I'm still using the old `AlgebraHomomorphism`. Should I clean this up before merging?

**Comment by thofma** [comment]
That would be great

**Comment by HechtiDerLachs** [comment]
I will rename the `canonically_isomorphic` routine. I also need to insert some type assertions. Sorry for not managing to do this today.

But the failing tests are not my fault, are they?

**Comment by thofma** [comment]
Yes, you can ignore them.

**Comment by HechtiDerLachs** [comment]
Just a heads up: This should be ready to  be merged now. The failing checks are for other reasons.

**Pull request merged**